### PR TITLE
support new configuration option `moduleNamingMode`

### PR DESCRIPTION
**Pull request by HunterLarco**
## TL;DR

Prior to this change, schema files had to be flattened within modules; otherwise, this plugin would break. This update introduces a new option, `moduleNamingMode`, that allows for deeply nested schema files. The change is backward compatible with existing workflows.

## Background

When parsing GraphQL schema files, the plugin identifies the "module name" for each schema file. It [currently determines the module name by taking the parent directory of each schema file](https://github.com/eddeee888/graphql-code-generator-plugins/blob/master/packages/typescript-resolver-files/src/parseSources/parseSources.ts#L28). This approach works well when the last folder denotes the module name, as seen in [nested-domain-modules](https://github.com/eddeee888/graphql-code-generator-plugins/blob/master/packages/typescript-resolver-files-e2e/src/test-nested-domain-modules).

## Problem

The current behavior causes issues when a project organizes schema files within additional subfolders inside a module. For example, our project structure looks like this:

```
codegen.ts
modules/
  user/
    schema/
      User.graphql
  profile/
    schema/
      Profile.graphql
      User.graphql
```

We configure `codegen.ts` to generate a `resolvers/` directory in each module because we value separating implementation from API definitions. However, this setup breaks the plugin. In the example above, the module name for both `profile/schema/User.graphql` and `user/schema/User.graphql` is incorrectly identified as `schema`. This happens because the plugin takes the parent directory (`schema`) as the module name.

As a result, the definitions are merged under the same module name `schema`, and `User` resolvers are only generated for one of the modules—depending on the order in which the files are processed—leading to unpredictable behavior.

## Solution

To address this issue, I've added a new option `moduleNamingMode` that allows clients to specify how the module name is determined:

- `last` (default): The module name is the last folder in a file's path (existing behavior).
- `first`: The module name is the first folder (within the working directory) in a file's path.

Using `first` means that if codegen is pointed to `/path/to/modules`, the module names will be the first layer of folders within `/path/to/modules` (e.g., `user` and `profile`). This allows for deeper nesting within modules without breaking the plugin.

Please refer to the e2e test included in this PR for an example of how this works in practice.

### CODE REVIEW

**Comment by eddeee888** [comment]
Love the idea @HunterLarco !

I wonder if this could be generalised further? 

One option I'm thinking is letting the user chose the index of folder from base dir to schema file. The default is `-1` which is the same as the `last` option. `0` would be the same as `first` option

For example, if `./modules` is the base dir:
- If the path is `./modules/user/schema.graphql`, then the relative path array is `["user"]`, then `-1` (or `0`) both work
- Some users may want to group modules based on the domain e.g. `./modules/user-domain/user/schema.graphql` and `./modules/user-domain/profile/schema.graphql`, 
  - then the relative path array are `["user-domain", "user"]`, `["user-domain", "profile"]` respectively
  - then `-1` will pick `user` and `profile` respectively as the module names
- Your use case of `./user/schema/schema.graphql` should work as well with `0`
- For other nested use cases e.g. `modules/foo/bar/baz/qux/User.graphql`, `2` can be used to target `baz` 

I _think_ we might be able to do `relativePathFromBaseToModule.at(theIndexConfig)` here

---

If we did go with the index approach, do you have suggestion for the option name? 🙂

**Comment by HunterLarco** [comment]
@eddeee888 that's a good question! I'm happy to expand the functionality here to be more flexible. Unfortunately I'm drawing a blank for naming. We could keep `moduleNamingMode` and simply explain the logic in the documentation. What do you think?

**Comment by eddeee888** [comment]
Yes, I think your suggestion works. I have plan to rename a few options for v1, so this rename can happen there if needed. Thank you!

**Comment by HunterLarco** [comment]
@eddeee888 apologies for the delay. I hope you had a nice holiday season. I've updated the PR (and documentation) to reflect your suggestion. Now in addition to `first` and `last`, clients may use any number positive or negative to select the module name from the schema's path. Let me know if there's anything else I can do to help this land :)

**Comment by eddeee888** [comment]
Thank you @HunterLarco ! There's only a small conflict so I'll fix and make sure CI passes to merge

**Comment by eddeee888** [comment]
Thank you! I've released an alpha for this feature: `@eddeee888/gcg-typescript-resolver-files@pr346-run668-1`
